### PR TITLE
Kernel: Make sure we free the thread stack and TLS region on thread exit

### LINKED ISSUE #7267: LibPthread: stack leaks after thread exits
Every time a process starts a thread using `pthread_create` without providing a stack in the attributes, a new 4 MiB stack is created:

https://github.com/SerenityOS/serenity/blob/598d7f412730cdec08510d551aabab93c8c7685e/Userland/Libraries/LibPthread/pthread.cpp#L99-L107

When the thread exits however, the `mmap`ed stack does not get `munmap`ed and 4 MiB of stack is leaked.

image

### PR DESCRIPTION
**Pull request by gunnarbeutner**
**LibPthread: Make some variables static**

**Kernel: Make sure we unmap the TLS region when a thread exits**

Previously the TLS region would get leaked which was noticible when creating/destroying a lot of threads and then inspecting
the region map.

**Kernel: Move sys$munmap functionality into a helper method**

**Kernel: Make sure we free the thread stack on thread exit**

This adds two new arguments to the thread_exit system call which let a thread unmap an arbitrary VM range on thread exit. LibPthread uses this functionality to unmap the thread stack.

Fixes #7267.

### CODE REVIEW

File: Kernel/Thread.cpp [hunk 1]
```diff
@@ -262,6 +262,12 @@ void Thread::exit(void* exit_value)
     set_should_die();
     u32 unlock_count;
     [[maybe_unused]] auto rc = unlock_process_if_locked(unlock_count);
+    if (m_thread_specific_range.has_value()) {
+        auto* region = process().space().find_region_from_range(m_thread_specific_range.value());
+        VERIFY(region);
+        if (!process().space().deallocate_region(*region))
+            dbgln("Failed to unmap TLS range, exiting thread anyway.");
```
Comment (bgianfo) [inline Kernel/Thread.cpp:269]: Should we include the error in the message?

Comment (gunnarbeutner) [reply Kernel/Thread.cpp:269]: Sure, the best I could do is include the error _number_ though. Any ideas? :)

Comment (bgianfo) [reply Kernel/Thread.cpp:269]: Error was all I was hoping for.

Comment (gunnarbeutner) [reply Kernel/Thread.cpp:269]: Oh, also, dealocate_region only returns a bool.

**Pull request merged**